dormeneur/To-Do-List
# mplement Dexie.js for Persistent Todo Management with Drag-and-Drop Support (Fixes #9)

This pull request introduces Dexie.js to manage the persistence of todos in the browser's IndexedDB, ensuring todos are saved and not deleted on page reload. Additionally, it enables the persistence of state changes after todos are dragged and dropped into different categories, fixing issue #9.

Key Features:
Dexie.js Integration:

Todos are now stored in the browser’s IndexedDB using Dexie.js, allowing them to persist across page reloads or browser restarts.
A new TodoDatabase has been created, which stores the todos with fields like id, title, and state.
Todo Creation:

Newly created todos are automatically saved in IndexedDB and assigned a default state (todo).
On reload, previously created todos are retrieved and displayed in their respective states.
Drag-and-Drop State Management:

Todos can now be dragged and dropped between different categories (e.g., from "todo" to "in-progress" or "done").
The state of each todo is updated in Dexie.js based on where it was dropped, ensuring the state change is persistent across reloads.
State Persistence:

After a todo is moved to a different category, the updated state is immediately saved to the database.
On reload, todos appear in the correct category corresponding to their saved state.
Issue #9 Fixed:

The bug where todos were not persistent after page reload and state changes were not saved is now resolved.
Todos and their states now remain consistent and reliable, addressing user feedback and improving UX.
Technical Changes:
Dexie.js Database Setup:
Created a TodoDatabase with a schema for todos including fields for id, title, and state.
State Update Logic:
Implemented the updateTodoState function to update the state of a todo after it is dropped into a new category.
Drag-and-Drop Handling:
Enhanced the drag-and-drop functionality to trigger state updates in Dexie.js and ensure these changes persist.
How to Test:
Create a new todo:

Add a new todo and confirm it is saved in IndexedDB by checking the browser's developer tools.
Reload the page and verify that the todo still appears in the UI.
Drag and drop todos:

Drag a todo to a new category (e.g., "in-progress" or "done").
Confirm that the todo's state is updated in the UI and Dexie.js.
Reload the page and ensure the todo remains in the correct category.
